libs/sysplugins/smarty_internal_runtime_writefile.php

```diff
@@ -29,20 +29,14 @@ public function writeFile($_filepath, $_contents, Smarty $smarty)
     {
         $_error_reporting = error_reporting();
         error_reporting($_error_reporting & ~E_NOTICE & ~E_WARNING);
-        $_file_perms = property_exists($smarty, '_file_perms') ? $smarty->_file_perms : 0644;
-        $_dir_perms =
```

Comment:
_dr_perms property does not exist, the __get method always returns null for this property as _dir_perms declared as obsoleted property